durabletask/worker.py

```diff
@@ -602,17 +673,37 @@ def process_event(self, ctx: _RuntimeOrchestrationContext, event: pb.HistoryEven
                         self._logger.warning(
                             f"{ctx.instance_id}: Ignoring unexpected subOrchestrationInstanceFailed event with ID = {task_id}.")
                     return
-                sub_orch_task.fail(
-                    f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
-                    failedEvent.failureDetails)
-                ctx.resume()
+                if isinstance(sub_orch_task, task.RetryableTask):
+                    if sub_orch_task._retry_policy is not None:
+                        if sub_orch_task.try_completion():
+                            sub_orch_task.fail(
+                                f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
+                                failedEvent.failureDetails)
+                            ctx.resume()
+                        else:
+                            next_delay = sub_orch_task.compute_next_delay()
+                            sub_orch_task.increment_retry_count()
+                            if next_delay == timedelta.min:
+                                sub_orch_task.fail(
+                                    f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
+                                    failedEvent.failureDetails)
+                                ctx.resume()
+                            else:
+                                ctx.create_timer_internal(next_delay, sub_orch_task)
+                elif isinstance(sub_orch_task, task.CompletableTask):
+                    sub_orch_task.fail(
+                        f"Sub-orchestration task #{task_id} failed: {failedEvent.failureDetails.errorMessage}",
+                        failedEvent.failureDetails)
+                    ctx.resume()
```

Comment:
It seems we've repeated this `sub_orch_task.fail(...)` and `ctx.resume()` code three times in this block. Can we refactor this at all to reduce the duplication? My earlier suggestion might reduce at least one of these.

Reply:
yeah, now it is there 2 times.